server/src/storage/datastore.ts

```diff
@@ -33,4 +34,27 @@ const getWithId = async (kind: string, id: string) => {
   };
 };
 
-export {getWithId};
+/**
+ * A Datastore wrapper that inserts a particular entity with the specified Kind.
+ * @param kind The Kind of the Entity
+ * @param data The data of the Entity to be added
+ */
+const add = async (kind: string, data: object) => {
+  const key = datastore.key(kind);
+  const entity = {key, data};
+  const [res] = await datastore.insert(entity);
+  return getIdFromCommitResponse(res);
```

Comment:
This makes your add function nullable. Instead of that can you assert id != null or throw some exception of it's null

Reply:
Okay!